src/contracts/middleware/BLSOperatorStateRetriever.sol

```diff
@@ -96,36 +96,44 @@ contract BLSOperatorStateRetriever {
         uint32 referenceBlockNumber, 
         bytes calldata quorumNumbers, 
         bytes32[] calldata nonSignerOperatorIds
-    ) external returns (CheckSignaturesIndices memory) {
+    ) external view returns (CheckSignaturesIndices memory) {
         IStakeRegistry stakeRegistry = registryCoordinator.stakeRegistry();
         CheckSignaturesIndices memory checkSignaturesIndices;
 
         checkSignaturesIndices.nonSignerQuorumBitmapIndices = registryCoordinator.getQuorumBitmapIndicesByOperatorIdsAtBlockNumber(referenceBlockNumber, nonSignerOperatorIds);
```

Comment:
IMO this could use some comments given the complexity of the operations. IIRC this is getting the overlap of the set (`quorumNumbers`, the union of all quorums that the operators corresponding to `nonSignerOperatorIds` belonged to at the `referenceBlockNumber`).
If this is correct can we add a description to this effect? If it's not then please clarify for me!

Reply:
Oh wait there's no overlap happening here, it's just getting the second of these set members.
should we be getting this overlap / do we perform this operation later?

Reply:
am thinking just `checkSignaturesIndices.nonSignerQuorumBitmapIndices = quorumNumbers | checkSignaturesIndices.nonSignerQuorumBitmapIndices`

Reply:
nevermind, `checkSignaturesIndices.nonSignerQuorumBitmapIndices` is itself an array, where each member is the index of the bitmap of the i-th operator in `nonSignerOperatorIds`, at `referenceBlockNumber`.
OK just going to say again this is complex enough to need documentation throughout